pyatlan/client/group.py

```diff
@@ -152,9 +152,10 @@ def get_all(
         :param columns: provides columns projection support for groups endpoint
         :returns: a list of all the groups in Atlan
         """
-        if response := self.get(offset=offset, limit=limit, sort=sort, columns=columns):
-            return response.records  # type: ignore
-        return None  # type: ignore
+        response: GroupResponse = self.get(
+            offset=offset, limit=limit, sort=sort, columns=columns
+        )
+        return response  # type: ignore
 
     @validate_arguments
     def get_by_name(
```

Comment:
Let's update this method as well, since it also returns a list of `AtlanGroup`. We should have it return a `GroupResponse` instead, which includes support for iterative pagination.